### PR TITLE
LW-7811(hardware-trezor) additional witness requests mapper

### PR DESCRIPTION
**Pull request by DominikGuzei**
# Context

This PR implements the `additionalWitnessRequests` required for multi-sig and and token minting witness requests: https://github.com/trezor/connect/blob/e92440e4af786bab7f326397b3e9a64e6c9df77c/docs/methods/cardanoSignTransaction.md?plain=1#L36 

it also fixes one broken unit test for the asset mapper.

### CODE REVIEW

File: packages/hardware-trezor/package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "@cardano-sdk/key-management": "workspace:~",
     "@cardano-sdk/tx-construction": "workspace:~",
     "@cardano-sdk/util": "workspace:~",
+    "lodash": "4.17.21",
```
Comment (marcin-mazurek) [inline packages/hardware-trezor/package.json:66]: How about more flexible versioning:
```suggestion
    "lodash": "^4.17.21",
```
We seem to be declaring dependencies like that everywhere else in SDK. Strict versions can easily lead to duplication of dependencies.

File: packages/hardware-trezor/src/transformers/additionalWitnessRequests.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as Trezor from 'trezor-connect';
+import { CardanoKeyConst, GroupedAddress, util } from '@cardano-sdk/key-management';
+import { TrezorTxTransformerContext } from '../types';
+import { isNotNil } from '@cardano-sdk/util';
+import concat from 'lodash/concat';
+import uniq from 'lodash/uniq';
+
+export const createRewardAccountKeyPath = (knownAddress: GroupedAddress) => [
```
Comment (marcin-mazurek) [inline packages/hardware-trezor/src/transformers/additionalWitnessRequests.ts:8]: Should that belong to `keyPaths.ts`? (just thinking out loud)

File: packages/hardware-trezor/src/transformers/additionalWitnessRequests.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  util.harden(CardanoKeyConst.PURPOSE),
+  util.harden(CardanoKeyConst.COIN_TYPE),
+  util.harden(knownAddress.accountIndex),
+  util.STAKE_KEY_DERIVATION_PATH.role,
+  util.STAKE_KEY_DERIVATION_PATH.index
+];
+
+export const mapAdditionalWitnessRequests = (inputs: Trezor.CardanoInput[], context: TrezorTxTransformerContext) => {
+  const paymentKeyPaths = uniq(inputs.map((i) => i.path).filter(isNotNil));
+  const additionalWitnessPaths = concat([], paymentKeyPaths);
```
Comment (marcin-mazurek) [inline packages/hardware-trezor/src/transformers/additionalWitnessRequests.ts:18]: How about just:
```suggestion
  const additionalWitnessPaths = [paymentKeyPaths];
```

File: packages/hardware-trezor/src/transformers/additionalWitnessRequests.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import concat from 'lodash/concat';
+import uniq from 'lodash/uniq';
+
+export const createRewardAccountKeyPath = (knownAddress: GroupedAddress) => [
+  util.harden(CardanoKeyConst.PURPOSE),
+  util.harden(CardanoKeyConst.COIN_TYPE),
+  util.harden(knownAddress.accountIndex),
+  util.STAKE_KEY_DERIVATION_PATH.role,
+  util.STAKE_KEY_DERIVATION_PATH.index
+];
```
Comment (tomislavhoracek) [inline packages/hardware-trezor/src/transformers/additionalWitnessRequests.ts:14]: Maybe we should use https://github.com/input-output-hk/cardano-js-sdk/blob/chore/lw-6514-hardware-trezor-package/packages/hardware-trezor/src/transformers/keyPaths.ts#L19 and include stakeKeyPath only if it exists?

File: packages/hardware-trezor/src/transformers/additionalWitnessRequests.ts (hunk 2 above)
Comment (tomislavhoracek) [reply packages/hardware-trezor/src/transformers/additionalWitnessRequests.ts:8]: @marcin-mazurek check the comment https://github.com/input-output-hk/cardano-js-sdk/pull/851#discussion_r1286943001 I think it answers your question

**Comment by DominikGuzei** [comment]
@tomislavhoracek @marcin-mazurek thanks for your feedback 🙏  all issues should be resolved now 👍

**Review by tomislavhoracek — APPROVED** [review]
@DominikGuzei 💯

**Comment by marcin-mazurek** [comment]
@DominikGuzei please run `yarn` locally and commit `yarn.lock` 🚓

**Comment by DominikGuzei** [comment]
@marcin-mazurek yarn lock file is updated 👍

**Review by marcin-mazurek — APPROVED** [review]

**Pull request merged**